packages/core/src/lib/persisted.ts

```diff
@@ -0,0 +1,88 @@
+import { writable, type Writable } from 'svelte/store';
+
+const stores: Record<string, Writable<unknown> | undefined> = {};
+
+const getStorage = (type: 'local' | 'session') => {
+  return type === 'local' ? localStorage : sessionStorage;
+};
+
+const createPersisted = <T>(
+  key: string,
+  initialValue: T | null,
+  storage: ReturnType<typeof getStorage>
+) => {
+  const store = writable<T | null>(initialValue, (set) => {
+    const json = storage.getItem(key);
+
+    if (json) {
+      try {
+        set(JSON.parse(json) as T);
+      } catch {
+        set(null);
+      }
```

Comment:
This safe parse is repeated in a few places, can it be moved to a function?

```ts
const parse = <T>(value: unknown): T => {
  try {
    return JSON.parse(value) as T;
  } catch {
    return null
  }
}
```

Reply:
Done!